### PR TITLE
Modal Status Bar Translucent

### PR DESCRIPTION
**Pull request by jemise111**
Currently the Modal component on Android is rendered below the Status Bar, which changes it's color to grey, and in the UIExplorer example the backdrop is just formatted to look the same color. In some scenarios users may want to preserve the color of their status bar and make it look as though the modal is appearing on top. This PR allows for that.

This GIF shows current behavior and new behavior with the translucentStatusBar prop set to true.

I've updated the UIExplorer app to demonstrate and the docs as shown below

image

Thanks!

### CODE REVIEW

File: Libraries/Modal/Modal.js [hunk 1]
```diff
@@ -20,6 +20,8 @@ const View = require('View');
 const requireNativeComponent = require('requireNativeComponent');
 const RCTModalHostView = requireNativeComponent('RCTModalHostView', null);
 
+const STATUS_BAR_HEIGHT = 24;
```
Comment (dmmiller) [inline Libraries/Modal/Modal.js]: Is this fixed on Android?

Comment (janicduplessis) [inline Libraries/Modal/Modal.js]: You can use StatusBar.currentHeight to get the actual height from native.

Comment (jemise111) [inline Libraries/Modal/Modal.js]: @janicduplessis thanks for the tip, fixed!

**Comment by janicduplessis** [comment]
Should we add a prop for this? Feels like it should always appear on top of the status bar. It would probably still be possible to achieve the old behavior using the `StatusBar` module and changing the status bar color when the modal is opened.

**Comment by janicduplessis** [comment]
Have you tested on older android version that do not support the TRANSLUCENT window flag?

**Comment by facebook-github-bot-7** [comment]
@jemise111 updated the pull request.

**Comment by facebook-github-bot-3** [comment]
@jemise111 updated the pull request.

**Comment by jemise111** [comment]
@janicduplessis Fixed so the top of the modal is only pushed down on the appropriate API level on android.

Nothing breaks in the Java code for older versions.

Could you please clarify your other comment? Thanks!

**Comment by janicduplessis** [comment]
I'm wondering if the modal should always render over the status bar (like when you set Translucent Statusbar to true in the example). I think it looks a lot better and is what pretty much every app does.

**Comment by satya164** [comment]
I think this should be the default behaviour instead of adding a prop.

**Comment by mkonicek** [comment]
Thanks for looking @janicduplessis and @grabbou. Let's make this default.

**Comment by jemise111** [comment]
@mkonicek great, will do! Think it's necessary to keep a prop for rendering the dialogue the original way? Something like 'persistsStatusBar={true}'?

**Comment by satya164** [comment]
@jemise111 I don't think anybody really wants that. Do you think people need it?

**Comment by jemise111** [comment]
@satya164 no, I agree with you. Just wanted to check.

**Comment by facebook-github-bot-6** [comment]
@jemise111 updated the pull request.

**Comment by jemise111** [comment]
@mkonicek done!

**Comment by facebook-github-bot-4** [comment]
@jemise111 updated the pull request.

**Comment by facebook-github-bot-9** [comment]
@jemise111 updated the pull request.

File: Libraries/Modal/Modal.js [hunk 2]
```diff
@@ -91,6 +103,18 @@ class Modal extends React.Component {
   }
 }
 
+Modal.propTypes = {
```
Comment (janicduplessis) [inline Libraries/Modal/Modal.js]: Why are these prop types added here? It's already defined in the class as a static member.

File: Examples/UIExplorer/ModalExample.js [hunk 3]
```diff
@@ -68,7 +68,7 @@ var ModalExample = React.createClass({
     return {
       animationType: 'none',
       modalVisible: false,
-      transparent: false,
```
Comment (janicduplessis) [inline Examples/UIExplorer/ModalExample.js]: nit: Keep the trailling comma here.

File: Libraries/Modal/Modal.js [hunk 4]
```diff
@@ -60,6 +63,15 @@ class Modal extends React.Component {
       backgroundColor: this.props.transparent ? 'transparent' : 'white',
     };
 
+    let topValue = 0;
```
Comment (janicduplessis) [inline Libraries/Modal/Modal.js]: Let's merge this with containerBackgroundColor and remove the extra variables.
``` js
const containerStyles = {
  backgroundColor: this.props.transparent ? 'transparent' : 'white',
  top: Platform.OS === 'android' && Platform.Version >= 19 ? STATUS_BAR_HEIGHT : 0,
};

**Comment by jemise111** [comment]
@janicduplessis done!

**Comment by facebook-github-bot-4** [comment]
@jemise111 updated the pull request.

**Comment by dmmiller** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-4** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/932270526893932/int_phab) to review.

**Comment by dmmiller** [comment]
This is failing some internal tests.  I am working to get those fixed and have this landed.

As a bonus, this pull request fixed an internal issue that someone posted about yesterday.  I told them to try this patch and it solved the issue.  That's pretty awesome!

**Comment by jemise111** [comment]
@dmmiller thanks! Let me know if I can help in any way. 

And that's really cool! Glad to hear it.

**Comment by dmmiller** [comment]
I'm working on open sourcing the test that is failing so you can look into it :)  And I'll keep looking at it as well.

**Comment by dmmiller** [comment]
@jemise111 So the problem is that by introducing a dependency on StatusBar, we have to update ReactPackage to include StatusBarManager which some of our internal apps don't do.  Instead what I've done is expose a StatusBarHeight constant on Modal to use.  I should be landing it today with that change and you can see.

**Comment by jemise111** [comment]
Awesome @dmmiller. Thanks for the update.

**Comment by sauceypan** [comment]
Is this behaviour achievable currently?

**Pull request closed without merging**